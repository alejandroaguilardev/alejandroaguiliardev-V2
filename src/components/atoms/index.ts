export * from "./CardContent/CardContent";
export * from "./Copyright/Copyright";
export * from "./Curriculum/Curriculum";
export * from "./ListOption/ListOption";
export * from "./Logotype/Logotype";
export * from "./Hero/Hero";
export * from "./Networks/Networks";
export * from "./PageRoute/PageRoute";
export * from "./TitleSection/TitleSection";
