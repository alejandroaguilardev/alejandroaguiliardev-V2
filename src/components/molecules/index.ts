export * from './CardItem/CardItem'
export * from './Menu/Menu'
export * from './MenuResponsive/MenuResponsive'
export * from './SocialNetworks/SocialNetworks'