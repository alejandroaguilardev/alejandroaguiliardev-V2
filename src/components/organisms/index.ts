export * from "./About/About";
export * from "./ContainerSocialNetworks/ContainerSocialNetworks";
export * from "./Education/Education";
export * from "./Experience/Experience";
export * from "./Footer/Footer";
export * from "./Navbar/Navbar";
export * from "./Projects/Projects";
export * from "./Skills/Skills";
