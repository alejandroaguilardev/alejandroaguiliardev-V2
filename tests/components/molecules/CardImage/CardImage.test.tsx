import { render, screen, getByRole } from "@testing-library/react";
import { CardImage } from "../../../../src/components/molecules";

describe("CardImage Test", () => {
	const title = "title Test";
	const icon = "/img/banner.jpg";
	const size = "large";
	const link = "link";
	const repository = "repository";

	test("toMacthSnapshot", () => {
		const { container } = render(
			<CardImage
				title={title}
				icon={icon}
				size={size}
				link={link}
				repository={repository}
			/>
		);
		expect(container).toMatchSnapshot();
	});

	test("toMacthSnapshot", () => {
		render(
			<CardImage
				title={title}
				icon={icon}
				size={size}
				link={link}
				repository={repository}
			/>
		);

		const linkRole = screen.getByRole("link");
		const repositoryRole = screen.getByRole("repository");
		
		expect(linkRole.getAttribute("href")).toBe(link);
		expect(repositoryRole.getAttribute("href")).toBe(repository); 
	});
});
