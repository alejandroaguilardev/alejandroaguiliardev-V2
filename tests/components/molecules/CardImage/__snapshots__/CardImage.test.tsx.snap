// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImage Test toMacthSnapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1wx800i"
  >
    <span
      style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
    >
      <img
        alt="title Test"
        data-nimg="fill"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
    <div
      class="MuiBox-root css-lzhgz5"
    >
      <a
        href="link"
        rel="”noopener noreferrer"
        role="link"
        target="_blank"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-xlnruk-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-15esv0h-MuiTypography-root"
          >
            Link Web
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        href="repository"
        rel="”noopener noreferrer"
        role="repository"
        target="_blank"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-xlnruk-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-15esv0h-MuiTypography-root"
          >
            Repository
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
  </div>
</div>
`;
